_source/_posts/2021-12-16-cloud-native-java-microservices-with-istio.md

```diff
@@ -0,0 +1,580 @@
+
+{% img blog/cloud-native-java-microservices-with-istio/k8s-deployment.png alt:"JHipster JDL deployment" width:"900" %}{: .center-image }
+
+As you can see, there is also a bunch of useful commands printed on the console that you can use to do the deployment.
+
+### Update OIDC configuration
+
+At this point we have everything ready, but since we didn't add the Okta OIDC client secret to the **application.yml** files, we need to add them as Kubernetes secrets. Under **kubernetes/store** folder add a new file names **okta-oidc-secrets.yml** with below content. You should ignore this file from GIT for security reasons.
+
+```yml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: okta-oidc
+  namespace: jhipster
+  labels:
+    app: okta-oidc
+type: Opaque
+data:
+  oidc-client-secret: <base64-of-your-okta-client-secret>
+```
+
+To get base64 of your client secret run the following command. The base64 command will be available if you have openssl installed.
+
+```bash
+$ echo -n "your-okta-client-secret" | base64
+```
+
+If you didn't add the client-id and issuer-url to the application.yml files, you can add them here or in the deployment spec below as they are not secrets.
+
+Now open the **\*-deployment.yml** file in the **store**, **invoice**, **product** and **notification** folder and add the below to the **env** section of the **containers**.
+
+```yml
+- name: SPRING_SECURITY_OAUTH2_CLIENT_REGISTRATION_OIDC_CLIENT_SECRET
+  valueFrom:
+    secretKeyRef:
+      name: okta-oidc
+      key: oidc-client-secret
+```
+
+Here is the **kubernetes/store/store-deployment.yml** file with the change for reference.
+
+```yml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: store
+  namespace: jhipster
+spec:
+  ...
+  template:
+    ...
+    spec:
+      ...
+      containers:
+        - name: store-app
+          image: deepu105/store
+          env:
+            - name: SPRING_PROFILES_ACTIVE
+              value: prod
+            ...
+            - name: SPRING_SECURITY_OAUTH2_CLIENT_REGISTRATION_OIDC_CLIENT_SECRET
+              valueFrom:
+                secretKeyRef:
+                  name: okta-oidc
+                  key: oidc-client-secret
+          resources:
+            ...
+```
+
+### Deploy to GKE
+
+We ready to deploy now. First, we need to build and push the images to the registry. For this we can use the handy [JIB](https://github.com/GoogleContainerTools/jib) commands provided by JHipster. Navigate to each of the microservice folder and run the command below.
+
+```bash
+$ cd store && ./gradlew bootJar -Pprod jib -Djib.to.image=deepu105/store
+$ cd invoice && ./gradlew bootJar -Pprod jib -Djib.to.image=deepu105/invoice
+$ cd notification && ./gradlew bootJar -Pprod jib -Djib.to.image=deepu105/notification
+$ cd product && ./gradlew bootJar -Pprod jib -Djib.to.image=deepu105/product
```

Comment:
Remove $ so commands can be copy/pasted. Maybe even set the username as a variable first so there's only one line that needs to be changed by the person doing the tutorial?